includes/admin/load.php

```diff
@@ -393,3 +396,30 @@ static function ( $name ) {
 		);
 	}
 }
+
+/**
+ * Callback function that print plugin progress indicator script.
+ *
+ * @since n.e.x.t
+ */
+function perflab_print_plugin_progress_indicator_script() {
+	$js = <<<JS
+( function ( document ) {
+	document.addEventListener( 'DOMContentLoaded', function () {
+		document.addEventListener( 'click', function ( event ) {
+			if (
+				event.target.classList.contains(
+					'perflab-install-active-plugin'
+				)
+			) {
+				const target = event.target;
+				target.classList.add( 'updating-message' );
+				target.innerHTML = wp.i18n.__( 'Activating…' );
+			}
+		} );
+	} );
+} )( document );
+JS;
+
+	wp_print_inline_script_tag( $js );
```

Comment:
Best practice to not use `innerHTML` but rather `textContent`. This avoids the risk of injected markup.

Also, I don't believe the `'Activating…'` string will get translated here. I believe it will need to be set via PHP, but @swissspidy can confirm.

```suggestion
	$js_function = <<<JS
		function addPluginProgressIndicator( message ) {
			document.addEventListener( 'DOMContentLoaded', function () {
				document.addEventListener( 'click', function ( event ) {
					if (
						event.target.classList.contains(
							'perflab-install-active-plugin'
						)
					) {
						const target = event.target;
						target.classList.add( 'updating-message' );
						target.textContent = message;
					}
				} );
			} );
		}
JS;

	wp_print_inline_script_tag(
		sprintf(
			'( %s )( %s );',
			$js_function,
			wp_json_encode( __( 'Activating…', 'default' ) )
		),
		array( 'type' => 'module' )
	);
```

Reply:
Thanks for feedback. Yes that string is not translated in inline script so i update in https://github.com/WordPress/performance/pull/1190/commits/59dc0945245dfecac2e99f87eee0daf57590f37a